File: frontend-public/lib/types/session.ts
export interface ToolCall {
  id: string
  tool_name: string
  arguments: Record<string, unknown>
  result?: unknown
  status: "pending" | "running" | "success" | "error"
}

export interface ChatMessage {
  id: string
  session_id: string
  role: "user" | "assistant" | "system"
  content: string
  tool_calls: ToolCall[]
  created_at?: string
}

export interface ChatSession {
  id: string
  owner_id: string
  agent_id: string
  title: string
  created_at?: string
  updated_at?: string
}

export interface ChatSessionCreatePayload {
  agent_id: string
  title?: string
}

export interface ChatMessageCreatePayload {
  content: string
}

File: src/backend/core/shared/models/session.py
"""Session / Message 领域模型。"""

from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from typing import Any, Sequence


@dataclass
class ToolCall:
    """工具调用领域对象。"""

    id: str
    tool_name: str
    arguments: dict[str, Any]
    result: Any | None = None
    status: str = "pending"


@dataclass
class ChatMessage:
    """聊天消息领域对象。"""

    id: str
    session_id: str
    role: str
    content: str
    tool_calls: Sequence[ToolCall] = field(default_factory=list)
    created_at: datetime | None = None


@dataclass
class ChatSession:
    """聊天会话领域对象。"""

    id: str
    owner_id: str
    agent_id: str
    title: str
    created_at: datetime | None = None
    updated_at: datetime | None = None
